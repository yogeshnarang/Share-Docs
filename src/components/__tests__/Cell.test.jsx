// eslint-disable-next-line
import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Cell from '../Cell'

describe('A Cell', () => {
  const handleChangedCell = () => {}
  const executeFormula = () => {}
  const updateCells = () => {}

  const x = 4
  const y = 2
  const cell1 = shallow(
    <Cell
      key={`${x}-${y}`}
      y={y}
      x={x}
      onChangedValue={handleChangedCell}
      updateCells={updateCells}
      executeFormula={executeFormula}
      value={''}
    />,
  )

  const cell2 = shallow(
    <Cell
      key={`${x}-${y}`}
      y={y}
      x={x}
      onChangedValue={handleChangedCell}
      updateCells={updateCells}
      executeFormula={executeFormula}
      value={'Test'}
    />,
  )

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Cell
          key={`${x}-${y}`}
          y={y}
          x={x}
          onChangedValue={handleChangedCell}
          updateCells={updateCells}
          executeFormula={executeFormula}
          value={'Test'}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('is has empty content if initialized with value={""}', () => {
    expect(cell1.text()).toEqual('')
  })

  it('is contains "Test" if initialized with value={"Test"}', () => {
    expect(cell2.text()).toEqual('Test')
  })

  it('is a span in normal state', () => {
    expect(cell1.find('input').length === 0).toEqual(true)
    expect(cell1.find('span').length === 1).toEqual(true)
  })

  it('becomes an input if double clicked', () => {
    cell1.find('span').simulate('doubleClick')
    expect(cell1.find('span').length === 0).toEqual(true)
    expect(cell1.find('input').length === 1).toEqual(true)
  })
})
