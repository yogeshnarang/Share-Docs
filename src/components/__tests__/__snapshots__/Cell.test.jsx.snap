// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Cell renders correctly 1`] = `
<span
  onClick={[Function]}
  onDoubleClick={[Function]}
  role="presentation"
  style={
    Object {
      "border": "1px solid #cacaca",
      "boxSizing": "border-box",
      "color": "black",
      "display": "inline-block",
      "fontFamily": "Calibri, 'Segoe UI', Thonburi, Arial, Verdana, sans-serif",
      "fontSize": "14px",
      "height": "25px",
      "lineHeight": "15px",
      "margin": "0",
      "overflow": "hidden",
      "padding": "4px",
      "position": "relative",
      "textAlign": "left",
      "verticalAlign": "top",
      "width": "80px",
    }
  }
>
  Test
</span>
`;
